apps/happy-blocks/index.php

```diff
@@ -97,13 +97,45 @@ function a8c_happyblocks_view_assets() {
 add_action( 'wp_enqueue_scripts', 'a8c_happyblocks_view_assets' );
 
 /**
- * Get the topic's selected help site's domain.
+ * Get the domain to use in the Pricing Plans block.
  *
- * @return string
+ * @return string The domain
  */
-function a8c_happyblocks_get_topic_domain() {
-	$topic_id = bbp_get_topic_id();
-	return get_post_meta( $topic_id, 'which_blog_domain', true );
+function a8c_happyblocks_pricing_plans_get_domain() {
+
+	// If the user is not authenticated, then we can't get their domain.
+	if ( ! is_user_logged_in() ) {
+		return null;
+	}
+
+	$topic_id  = bbp_get_topic_id();
+	$author_id = get_post_field( 'post_author', $topic_id );
+
+	/*
+	If the current user is the author of the topic, return the topic's domain selected
+	in the "Site you need help with" field.
+	*/
+	if ( get_current_user_id() === $author_id ) {
+		$topic_domain = get_post_meta( $topic_id, 'which_blog_domain', true );
+		if ( $topic_domain ) {
+			return $topic_domain;
+		}
+	}
+
+	/*
+	If the user is logged in but not the author of the topic,
+	return the domain of the user's primary blog.
+	*/
+	$user = wp_get_current_user();
+
+	if ( isset( $user->primary_blog ) ) {
+		$primary = get_blog_details( $user->primary_blog );
+		if ( $primary ) {
+			return $primary->domain;
+		}
+	}
+
+	return null;
```

Comment:
Perhaps we could return `false` here and check for that later on?

Reply:
I would return `''`, I like it when functions return consistent types

Reply:
> I was using null as a way to mean "there's no valid value to show".

I agree that we need a way to signify that there's no valid value to show, but I think we should use `null` or `false` and not an empty string. I'd return `false` as `null` is usually used for returning a problem if the non-problematic return is an object. I know string is also an object in JS, but probably not mentally. In Calypso, for all the state selectors we have, we return `null` in case of a problem and otherwise the state subtree object. I'd return `false` here as I mentioned due to the fact that the correct return is a string, but `null` would work too for me.

Reply:
> I would return '', I like it when functions return consistent types

I suppose that's what you want to avoid in case of a problem: returning a consistent type with the case of a non-problem. That's why `null` or `false`.